src/test/java/teammates/test/cases/action/AdminEmailLogPageActionTest.java

```diff
@@ -0,0 +1,425 @@
+ * can be solved by rerunning the test cases at a different time.
+ */
+public class AdminEmailLogPageActionTest extends BaseActionTest {
+
+    // The test data will be a List<List<String>>. These constants are indexes
+    // for the list of log messages in the outer list.
+    private static final int LOG_MESSAGE_INDEX_TODAY = 0;
+    private static final int LOG_MESSAGE_INDEX_YESTERDAY = 1;
+    private static final int LOG_MESSAGE_INDEX_TWO_DAYS_AGO = 2;
+    private static final int LOG_MESSAGE_INDEX_MANY_LOGS = 3;
+
+    // In the case of many logs, the query will first look at logs within 2 hours before now,
+    // if the number of logs exceeds 50, it will stop the query and return logs from the first 2 hours.
+    // 130 seconds is chosen so that it will be around 50 logs within 2 hours before now.
+    private static final int LOG_MESSAGE_INTERVAL_MANY_LOGS = 130;
+
+    private List<List<String>> logMessages;
+
+    private SimpleDateFormat formatterAdminTime;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ActionURIs.ADMIN_EMAIL_LOG_PAGE;
+    }
+
+    @Override
+    protected void prepareTestData() {
+        loadLogMessages();
+        initVariable();
+    }
+
+    private void loadLogMessages() {
+        try {
+            String pathToJsonFile = TestProperties.TEST_DATA_FOLDER + "/typicalEmailLogMessage.json";
+            String jsonString = FileHelper.readFile(pathToJsonFile);
+            Type listType = new TypeToken<List<List<String>>>(){}.getType();
+
+            logMessages = JsonUtils.fromJson(jsonString, listType);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void initVariable() {
+        formatterAdminTime = new SimpleDateFormat("dd/MM/yy");
+        formatterAdminTime.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
+    }
+
+    @Override
+    @Test
+    public void testExecuteAndPostProcess() {
+        // See each independent test case
+    }
+
+    @BeforeGroups("typicalEmailLogs")
+    public void removeAndRestoreLogMessage() {
+        gaeSimulation.loginAsAdmin("admin");
+        gaeSimulation.clearLogs();
+
+        Date twoDaysAgo = TimeHelper.getDateOffsetToCurrentTime(-2);
+        insertLogMessagesAtTime(logMessages.get(LOG_MESSAGE_INDEX_TWO_DAYS_AGO), twoDaysAgo.getTime());
+        Date yesterday = TimeHelper.getDateOffsetToCurrentTime(-1);
+        insertLogMessagesAtTime(logMessages.get(LOG_MESSAGE_INDEX_YESTERDAY), yesterday.getTime());
+        Date today = TimeHelper.getDateOffsetToCurrentTime(0);
+        insertLogMessagesAtTime(logMessages.get(LOG_MESSAGE_INDEX_TODAY), today.getTime());
+    }
+
+    @Test(groups = "typicalEmailLogs")
+    public void filterQuery_invalidQuery_defaultSearchPerformed() {
+        int[][] expected = new int[][]{{0, 1, 2, 3}};
+        String query = "unknown";
+        verifyActionResult(expected, "filterQuery", query);
```

Comment:
Not sure if it's better to inline query, e.g. `verifyActionResult(expected, "filterQuery", "unknown");`